src/lib/api.ts

```diff
@@ -27,6 +35,134 @@ export async function createGroup(groupFormValues: GroupFormValues) {
   })
 }
 
+export async function acquireLockRecurringTransaction(groupId:string, expenseId:string, lockId: string): Promise<RecurringTransactions[]> {
+  const prisma = await getPrisma()
+  let retryCnt = 5
+  let getRecTxn:RecurringTransactions[] = []
+  while(--retryCnt) {
+    const existingLock = await prisma.recurringTransactions.findMany({
+      where: {
+        groupId,
+        expenseId
+      }
+    })
+    if (!existingLock.length) {
+      const query = `
+        INSERT INTO "RecurringTransactions"
+          ("groupId", "expenseId", "createNextAt", "lockId")
+          VALUES
+          ('${groupId}', '${expenseId}', 0, '${lockId}')
+          ON CONFLICT("groupId", "expenseId")
+          DO NOTHING;
+      `
+      await prisma.$queryRawUnsafe(query)
```

Comment:
I don't know Prisma but I do not recommend building SQL queries via string interpolation. This is how SQL injection vulnerabilities arise, not the "unsafe" in the function name.

Reply:
Agreed. I searched for Prisma docs but there's no option for 'createIfNotExists' hence used this method. I need this functionality for avoiding race conditions in between of inserting and fetching.

I also don't like building SQL queries like this - can you please help here for the alternatives for the above requirement?

Reply:
Could something like this work? [reference](https://www.prisma.io/docs/orm/prisma-client/using-raw-sql/raw-queries#queryraw). I'm not following the issue with `createIfNotExists`.

```ts
const query = await prisma.$queryRaw`
  INSERT INTO "RecurringTransactions"
    ("groupId", "expenseId", "createNextAt", "lockId")
  VALUES
    (${groupId}, ${expenseId}, 0, ${lockId})
  ON CONFLICT ("groupId", "expenseId")
  DO NOTHING;
`
```

Reply:
You mean to use `queryRaw` instead of `queryRawUnsafe`?